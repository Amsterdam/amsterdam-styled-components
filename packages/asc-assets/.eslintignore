src/**/*.ts
node_modules
