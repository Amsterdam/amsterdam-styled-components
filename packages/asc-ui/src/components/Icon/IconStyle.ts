import styled, { css, Theme } from '@datapunt/asc-core'
import { svgFill } from '../../utils'

export type Props = {
  inline?: boolean
  color?: Theme.ColorType
  iconUrl?: string
  size?: number
  padding?: number
  rotate?: number
}

export const defaultProps = {
  size: 20,
  padding: 0,
  rotate: 0,
}

const IconStyle = styled.span<Props>`
  display: ${({ inline }) => (inline ? 'inline-flex' : 'flex')};
  ${({ iconUrl }) => iconUrl && `background-image: ${iconUrl}`}
  ${({ size = defaultProps.size, padding = defaultProps.padding }) => css`
    width: ${size - padding * 2}px;
    height: ${size - padding * 2}px;
  `}
  ${({ padding }) =>
    padding &&
    css`
      padding: ${padding}px;
    `};
  box-sizing: content-box;
  ${({ rotate = defaultProps.rotate }) => `transform: rotate(${rotate}deg)`};

  & > svg {
    ${({ size = defaultProps.size, padding = defaultProps.padding }) => css`
      width: ${size - padding * 2}px;
      height: ${size - padding * 2}px;
    `}
  }
  
  ${({ color, theme }) => svgFill(color)({ theme })};
`

export default IconStyle
