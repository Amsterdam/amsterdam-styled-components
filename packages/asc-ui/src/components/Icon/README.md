# Icons List

room for icons list here
