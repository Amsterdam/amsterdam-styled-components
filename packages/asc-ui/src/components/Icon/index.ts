export { default as IconStyle } from './IconStyle'
export { default } from './Icon'
