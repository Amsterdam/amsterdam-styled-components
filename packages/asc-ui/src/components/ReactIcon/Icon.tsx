import * as React from 'react'
import {
  AbstractNode,
  IconDefinition,
} from '@datapunt/asc-assets/lib/Icons/types'
import { generate, isIconDefinition, log, MiniMap } from '../../utils/utils'

export interface IconProps {
  type: IconDefinition
  className?: string
  onClick?: React.MouseEventHandler<SVGSVGElement>
  style?: React.CSSProperties
  focusable?: string
}

class Icon extends React.Component<IconProps> {
  static displayName = 'IconReact'

  static definitions = new MiniMap<IconDefinition>()

  static add(...icons: IconDefinition[]) {
    icons.forEach(icon => {
      this.definitions.set(icon.name, icon)
    })
  }

  static clear() {
    this.definitions.clear()
  }

  static get(key?: string): IconDefinition | undefined {
    if (key) {
      const target = this.definitions.get(key)
      return target
    }
    return undefined
  }

  render() {
    const { type, ...rest } = this.props

    let target: IconDefinition | undefined
    if (isIconDefinition(type)) {
      target = type
    } else if (typeof type === 'string') {
      target = Icon.get(type)
      if (!target) {
        return null
      }
    }
    if (!target) {
      log(`type should be string or icon definiton, but got ${type}`)
      return null
    }
    return generate(target.icon as AbstractNode, `svg-${target.name}`, {
      fill: 'currentColor',
      focusable: 'false',
      ...rest,
    })
  }
}

export default Icon
