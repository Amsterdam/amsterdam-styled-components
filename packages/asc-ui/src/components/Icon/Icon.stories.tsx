import { Location, Passport } from '@datapunt/asc-assets'
import React from 'react'
import Icon from './Icon'

export default {
  title: 'Experimental/Atoms/Icon',
  decorators: [
    (storyFn: () => React.ReactNode) => (
      <div
        style={{
          display: 'flex',
          maxWidth: 400,
          flexDirection: 'column',
          alignItems: 'flex-start',
          padding: '40px 10px',
        }}
      >
        {storyFn()}
      </div>
    ),
  ],
}

export const Default = () => (
  <Icon>
    <Location />
  </Icon>
)

export const InlineFlex = () => (
  <Icon inline>
    <Location />
  </Icon>
)

export const Sizes = () => (
  <>
    <Icon size={14}>
      <Location />
    </Icon>
    Small
    <div style={{ paddingTop: 20 }}>
      <Icon size={20}>
        <Location />
      </Icon>
      Medium
    </div>
    <div style={{ paddingTop: 20 }}>
      <Icon size={32}>
        <Location />
      </Icon>
      Large
    </div>
    <div style={{ paddingTop: 20 }}>
      <Icon size={40}>
        <Location />
      </Icon>
      ExtraLarge
    </div>
  </>
)

export const ColorError = () => (
  <>
    <Icon inline size={40} color="error">
      <Passport />
    </Icon>
  </>
)

export const Rotation = () => (
  <>
    <Icon size={40}>
      <Location />
    </Icon>
    0 deg
    <div style={{ paddingTop: 20 }}>
      <Icon size={40} rotate={90}>
        <Location />
      </Icon>
      90 deg
    </div>
    <div style={{ paddingTop: 20 }}>
      <Icon size={40} rotate={180}>
        <Location />
      </Icon>
      180 deg
    </div>
    <div style={{ paddingTop: 20 }}>
      <Icon size={40} rotate={270}>
        <Location />
      </Icon>
      270 deg
    </div>
    <div style={{ paddingTop: 20 }}>
      <Icon size={40} rotate={68}>
        <Location />
      </Icon>
      68 deg (any number)
    </div>
  </>
)
