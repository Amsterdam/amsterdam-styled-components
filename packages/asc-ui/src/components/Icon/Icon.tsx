import IconStyle, { defaultProps, Props as IconStyleProps } from './IconStyle'

export { defaultProps }

export type Props = IconStyleProps

export default IconStyle
