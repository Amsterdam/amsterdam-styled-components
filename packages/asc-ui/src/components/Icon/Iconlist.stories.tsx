import React from 'react'
import * as Icons from '@datapunt/asc-assets'
import styled from 'styled-components'
import Icon from './Icon'

const IconWrapper = styled.span`
  display: inline-block;
  width: 170px;
  height: 170px;
`
const IconName = styled.span`
  display: block;
`

export default {
  title: 'Experimental/Atoms/IconList',
  decorators: [
    (storyFn: () => React.ReactNode) => (
      <div
        style={{
          display: 'inline-block',
          padding: '40px 10px',
        }}
      >
        {storyFn()}
      </div>
    ),
  ],
}

let CustomIcon
export const IconList = () => (
  <>
    {Object.entries(Icons).map((IconObject) => (
      <IconWrapper key={IconObject[1]} style={{}}>
        {(CustomIcon = IconObject[1])}
        <Icon size={100} inline>
          <CustomIcon />
        </Icon>
        <IconName>{IconObject[0]}</IconName>
      </IconWrapper>
    ))}
  </>
)
