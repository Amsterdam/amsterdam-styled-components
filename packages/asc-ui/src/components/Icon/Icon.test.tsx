import { Location } from '@amsterdam/asc-assets'
import { render } from '@testing-library/react'
import Icon from './Icon'

describe('Icon', () => {
  it('should render the location icon', () => {
    const { container } = render(
      <Icon inline size={24} padding={4} color="red">
        <Location />
      </Icon>,
    )
    expect(container.firstChild).toBeDefined()
  })
})
