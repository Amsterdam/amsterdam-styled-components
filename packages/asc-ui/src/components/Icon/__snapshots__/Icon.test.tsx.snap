// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should render the icon 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  width: 16px;
  height: 16px;
  padding: 4px;
  box-sizing: content-box;
  -webkit-transform: rotate(0deg);
  -ms-transform: rotate(0deg);
  transform: rotate(0deg);
}

.c0 > svg {
  width: 16px;
  height: 16px;
}

<span
  class="c0"
  size="24"
>
  icon-content
</span>
`;
